sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -17,47 +17,132 @@
   // the number of partitions
-  val numPartitions = parameters.getOrElse("numPartitions", null)
-
+  val numPartitions = parameters.getOrElse(JDBC_NUM_PARTITIONS, null)
   require(partitionColumn == null ||
     (lowerBound != null && upperBound != null && numPartitions != null),
-    "If 'partitionColumn' is specified then 'lowerBound', 'upperBound'," +
-      " and 'numPartitions' are required.")
+    s"If '$JDBC_PARTITION_COLUMN' is specified then '$JDBC_LOWER_BOUND', '$JDBC_UPPER_BOUND'," +
+      s" and '$JDBC_NUM_PARTITIONS' are required.")
+  val fetchSize = {
+    val size = parameters.getOrElse(JDBC_BATCH_FETCH_SIZE, "0").toInt
+    require(size >= 0,
+      s"Invalid value `${size.toString}` for parameter " +
+        s"`$JDBC_BATCH_FETCH_SIZE`. The minimum value is 0. When the value is 0, " +
+        "the JDBC driver ignores the value and does the estimates.")
+    size
+  }
 
   // ------------------------------------------------------------
-  // The options for DataFrameWriter
+  // Optional parameters only for writing
   // ------------------------------------------------------------
   // if to truncate the table from the JDBC database
-  val isTruncate = parameters.getOrElse("truncate", "false").toBoolean
+  val isTruncate = parameters.getOrElse(JDBC_TRUNCATE, "false").toBoolean
   // the create table option , which can be table_options or partition_options.
   // E.g., "CREATE TABLE t (name string) ENGINE=InnoDB DEFAULT CHARSET=utf8"
   // TODO: to reuse the existing partition parameters for those partition specific options
-  val createTableOptions = parameters.getOrElse("createTableOptions", "")
+  val createTableOptions = parameters.getOrElse(JDBC_CREATE_TABLE_OPTIONS, "")
+  val batchSize = {
+    val size = parameters.getOrElse(JDBC_BATCH_INSERT_SIZE, "1000").toInt
+    require(size >= 1,
+      s"Invalid value `${size.toString}` for parameter " +
+        s"`$JDBC_BATCH_INSERT_SIZE`. The minimum value is 1.")
+    size
+  }
+  val isolationLevel =
+    parameters.getOrElse(JDBC_TXN_ISOLATION_LEVEL, "READ_UNCOMMITTED") match {
+      case "NONE" => Connection.TRANSACTION_NONE
+      case "READ_UNCOMMITTED" => Connection.TRANSACTION_READ_UNCOMMITTED
+      case "READ_COMMITTED" => Connection.TRANSACTION_READ_COMMITTED
+      case "REPEATABLE_READ" => Connection.TRANSACTION_REPEATABLE_READ
+      case "SERIALIZABLE" => Connection.TRANSACTION_SERIALIZABLE
+    }
+}
+
+object JDBCOptions {
+  val JDBC_URL = "url"
+  val JDBC_TABLE_NAME = "dbtable"
+  val JDBC_DRIVER_CLASS = "driver"
+  val JDBC_PARTITION_COLUMN = "partitionColumn"
+  val JDBC_LOWER_BOUND = "lowerBound"
+  val JDBC_UPPER_BOUND = "upperBound"
+  val JDBC_NUM_PARTITIONS = "numPartitions"
+  val JDBC_BATCH_FETCH_SIZE = "fetchsize"
+  val JDBC_TRUNCATE = "truncate"
+  val JDBC_CREATE_TABLE_OPTIONS = "createTableOptions"
+  val JDBC_BATCH_INSERT_SIZE = "batchsize"
+  val JDBC_TXN_ISOLATION_LEVEL = "isolationLevel"
+
+  private val jdbcOptionNames = Seq(
```

Comment:
Using `Enumeration` looks like an overkill, and using reflection may be too hacky. One possible solution is to follow SQLConf:
```
object JDBCOptions {
  private val jdbcOptionNames = ArrayBuffer.empty[String]

  private def newOption(name: String): String = {
    jdbcOptionNames += name
    name
  }

  val JDBC_URL = newOption("url")
  ...
}
```
I'm ok to maintains this list manually, if it's hard to make it automatic